internal/sockstate/netstat_linux.go

```diff
@@ -278,20 +278,18 @@ func tcpSocks(accept AcceptFn) ([]sockTabEntry, error) {
 	return allTabs, nil
 }
 
-// GetConnInode returns the Linux inode number of a TCP connection
-func GetConnInode(c *net.TCPConn) (n uint64, err error) {
-	if isWSL || isProcBlocked {
-		return 0, ErrSocketCheckNotImplemented.New()
-	}
-
-	f, err := c.File()
+// GetConnInode returns the inode number of an fd.
+func GetConnInode(conn *net.TCPConn) (n uint64, err error) {
+	fd, err := getConnFd(conn)
 	if err != nil {
-		return
+		return 0, err
 	}
 
-	defer f.Close()
+	if isWSL || isProcBlocked {
```

Comment:
Seems like this check still belongs above the one above, no? Otherwise we get a less useful error message on WSL

I can test this a bit later

Reply:
Wait, surely TCPConn.File() on WSL doesn't fail? I was trying to mimic the behavior, same as on macOS, where we ensure the ofile got put back into non-blocking.

Reply:
Glad I tested this, it's definitely broken on WSL!

Digging into it, will have some patches a bit later.

Reply:
Nevermind! Turns out this was hitting a different unrelated bug, this appears to work fine!